include/zephyr/devicetree.h

```diff
@@ -917,6 +917,226 @@
 #define DT_STRING_TOKEN(node_id, prop) \
 	DT_CAT4(node_id, _P_, prop, _STRING_TOKEN)
 
+/**
+ * @brief Get a string property's value as a token.
+ *
+ * This removes "the quotes" from a string property's value,
+ * converting any non-alphanumeric characters to underscores. This can
+ * be useful, for example, when programmatically using the value to
+ * form a C variable or code.
+ *
+ * DT_STRING_TOKEN_PREFIX() can only be used for properties with string type.
+ *
+ * It is an error to use DT_STRING_TOKEN_PREFIX() in other circumstances.
+ *
+ * Example devicetree fragment:
+ *
+ * @code{.dts}
+ *     n1: node-1 {
+ *             prop = "foo";
+ *     };
+ *     n2: node-2 {
+ *             prop = "BAR";
+ *     }
+ *     n3: node-3 {
+ *             prop = "foo BAR";
+ *     };
+ * @endcode
+ *
+ * Example bindings fragment:
+ *
+ * @code{.yaml}
+ *     properties:
+ *       prop:
+ *         type: string
+ *         enum:
+ *           - "foo"
+ *           - "BAR"
+ *           - "foo BAR"
+ * @endcode
+ *
+ * Example enumeration
+ *
+ * @code{.c}
+ *     enum prop_value {
+ *         PROP_VAL_foo = 0x10,
+ *         PROP_VAL_BAR = 0x20,
+ *         PROP_VAL_foo_BAR = 0x40,
+ *     };
+ * @endcode
+ *
+ * Example usage:
+ *
+ * @code{.c}
+ *     DT_STRING_TOKEN_PREFIX(DT_NODELABEL(n1), prop, PROP_VAL_) // PROP_VAL_foo
+ *     DT_STRING_TOKEN_PREFIX(DT_NODELABEL(n2), prop, PROP_VAL_) // PROP_VAL_BAR
+ *     DT_STRING_TOKEN_PREFIX(DT_NODELABEL(n3), prop, PROP_VAL_) // PROP_VAL_foo_BAR
+ * @endcode
+ *
+ * Notice how:
+ *
+ * - The uppercased `"FOO"` in the DTS remains `FOO` as a token. It is
+ *   *not* converted to `foo`.
+ *
+ * - The whitespace in the DTS `"123 foo"` string is converted to
+ *   `123_foo` as a token.
+ *
+ * @param node_id node identifier
+ * @param prop lowercase-and-underscores property name
+ * @param prefix Prefix prepended to the string token
+ * @return the value of @p prop as a token, i.e. without any quotes
+ *         and with special characters converted to underscores
+ */
+#define DT_STRING_TOKEN_PREFIX(node_id, prop, prefix) \
+	UTIL_CAT(prefix, DT_STRING_TOKEN(node_id, prop))
```

Comment:
We use DT_CAT<x>, not UTIL_CAT, for reasons documented at the end of the file where the DT_CAT<x> macros are defined.

Reply:
Thanks for the review. If I were using `DT_CAT` I'd need to wrap these macros, because `DT_STRING_TOKEN` needs to expand first, which would not happen if I'd simply replaced `UTIL_CAT` with `DT_CAT`.

`UTIL_CAT` is already used on other occasions in this file, so I was thinking it's better to use it, instead of wrapping it.

Reply:
OK. Can you address the other comment?